### DISCUSSION TITLE
How to support strings with spaces and single quotes

### DISCUSSION THREAD

**Discussion by Quadrant6630** [junegunn/fzf#3532] [Q&A] [answered]
I have a simple process to look at the lines of a file, pass them to fzf for selection and then pass them to a script for processing. It currently looks like this: 

`cat $HOME/$PDA_FILE_DIR/todos/current | fzf | xargs -I {} ruby $HOME/$PDA_CONFIG_DIR/todo.rb delete current {}`

This works fine for strings that just have spaces such as "Clean the cat litter" and even double quotes such as "Clean the "cat" litter". But it fails with the error "xargs: unterminated quote" if there is single quote like "Clean the cat's litter". 

Any suggestions on how to make this work with lines that contain single quotes? Thanks in advance.

**Comment by LangLangBart** [comment, marked as answer]
- maybe this will work
```bash
cat current.txt | fzf | tr '\n' '\0' | xargs -0 -I {}
```

- alternatively, fzf can also read `NUL`
```bash
cat current.txt | tr '\n' '\0' | fzf --read0 | xargs -0 -I {}
```

EDIT
- Additionally, fzf can print `NUL`, making the `tr` command redundant.
```bash
cat current.txt | fzf --print0 | xargs -0 -I {}
```

**Reply by Quadrant6630** [reply]
Perfect. I tried both the first and third suggestions and both worked. Thank you very much.
